### PR TITLE
Add docker_termination_logging flag

### PR DESCRIPTION
**Pull request by ulzha**
This is a kludge to work around the linked [Docker bug](https://github.com/kubernetes/kubernetes/issues/41516).

### CODE REVIEW

File: styx-common/src/main/java/com/spotify/styx/model/DataEndpoint.java [hunk 1]
```diff
@@ -59,10 +70,12 @@ public static DataEndpoint create(
       @JsonProperty("partitioning") Partitioning partitioning,
       @JsonProperty("docker_image") Optional<String> dockerImage,
       @JsonProperty("docker_args") Optional<List<String>> dockerArgs,
+      @JsonProperty("docker_termination_logging") Optional<Boolean> dockerTerminationLogging,
       @JsonProperty("secret") Optional<Secret> secret,
       @JsonProperty("resources") List<String> resources) {
 
-    return new AutoValue_DataEndpoint(id, partitioning, dockerImage, dockerArgs, secret,
+    return new AutoValue_DataEndpoint(id, partitioning, dockerImage, dockerArgs,
+        dockerTerminationLogging.isPresent() && dockerTerminationLogging.get(), secret,
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/model/DataEndpoint.java]: `dockerTerminationLogging.orElse(false)` maybe

Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/model/DataEndpoint.java]: +1 for orElse

File: styx-common/src/main/java/com/spotify/styx/model/DataEndpoint.java [hunk 2]
```diff
@@ -47,6 +48,16 @@
...
 
+  /**
+   * Toggles behavior to reliably report exit status from the Docker container, via
+   * https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/#writing-and-reading-a-termination-message
+   *
+   * <p>Ideally this should be unneeded, but mere exitCode is known to sometimes spuriously
+   * return 0 when in fact the container has been killed. See https://github.com/kubernetes/kubernetes/issues/41516
+   */
+  @JsonProperty
+  public abstract Boolean dockerTerminationLogging();
```
Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/model/DataEndpoint.java]: `boolean` instead of `Boolean`

**Comment by danielnorberg** [comment]
👍

**Pull request merged**